new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -480,47 +619,139 @@ private String createImageAndVerify() {
+                    .includeServerOutInPodLog(true)
+                    .serverStartPolicy("IF_NEEDED")
+                    .serverPod(new ServerPod()
+                            .addEnvItem(new V1EnvVar()
+                                    .name("JAVA_OPTIONS")
+                                    .value("-Dweblogic.StdoutDebugEnabled=false"))
+                            .addEnvItem(new V1EnvVar()
+                                    .name("USER_MEM_ARGS")
+                                    .value("-Djava.security.egd=file:/dev/./urandom ")))
+                    .adminServer(new AdminServer()
+                            .serverStartState("RUNNING")
+                            .adminService(new AdminService()
+                                    .addChannelsItem(new Channel()
+                                            .channelName("default")
+                                            .nodePort(0))))
+                    .addClustersItem(new Cluster()
+                            .clusterName("cluster-1")
+                            .replicas(replicaCount)
+                            .serverStartState("RUNNING"))
+                    .configuration(new Configuration()
+                            .model(new Model()
+                                    .domainType("WLS")
+                                    .runtimeEncryptionSecret(encryptionSecretName))));
+
+    logger.info("Create domain custom resource for domainUID {0} in namespace {1}",
+            domainUID, domNamespace);
+    boolean domCreated = assertDoesNotThrow(() -> createDomainCustomResource(domain),
+            String.format("Create domain custom resource failed with ApiException for %s in namespace %s",
+                    domainUID, domNamespace));
+    assertTrue(domCreated, String.format("Create domain custom resource failed with ApiException "
+                    + "for %s in namespace %s", domainUID, domNamespace));
+  }
+
+  private void checkPodCreated(String podName, String domainUid, String domNamespace) {
     withStandardRetryPolicy
         .conditionEvaluationListener(
             condition -> logger.info("Waiting for pod {0} to be created in namespace {1} "
                     + "(elapsed time {2}ms, remaining time {3}ms)",
                 podName,
-                domainNamespace,
+                domNamespace,
                 condition.getElapsedTimeInMS(),
                 condition.getRemainingTimeInMS()))
-        .until(assertDoesNotThrow(() -> podExists(podName, domainUid, domainNamespace),
+        .until(assertDoesNotThrow(() -> podExists(podName, domainUid, domNamespace),
             String.format("podExists failed with ApiException for %s in namespace in %s",
-                podName, domainNamespace)));
+                podName, domNamespace)));
 
   }
 
-  private void checkPodRunning(String podName) {
+  private void checkPodReady(String podName, String domainUid, String domNamespace) {
```

Comment:
you mentioned in the comments that Xian is adding check for pod ready and here you changed the method name to checkPodReady?

Reply:
Mark asked if we were checking ready or running - they are both different. We should be checking ready. So, I changed the name to reflect that. But the actual TestAssertion method right now checks if pod is running - not if pod is ready. That is what Xian is fixing.